gui/src/pages/NewProjectConfig.svelte

```diff
@@ -0,0 +1,156 @@
+<script lang="ts">
+  import Layout from '../components/Layout.svelte';
+  import Textbox from '../components/Textbox.svelte';
+  import Spinner from '../components/Spinner.svelte';
+  import ErrorLabel from '../components/ErrorLabel.svelte';
+  import SubmitButton from '../components/form/SubmitButton.svelte';
+  import CenterFormPanel from '../components/form/CenterFormPanel.svelte';
+  import type { ReadDirResult } from '../lib/api';
+  import { api, isClientError } from '../lib/api';
+  import * as router from 'svelte-spa-router';
+
+  export let params = { starter: '' };
+
+  const { starter } = params;
+
+  let name = starter;
+  let templateUrl: string | null = null;
+  let error: Error | null = null;
+
+  let workingDirectory: string | null = null;
+
+  const setWorkingDirectory = (dir: string) => {
+    workingDirectory = dir;
+    dirPromise = api.kernel.readDir(workingDirectory);
+  };
+
+  let dirPromise: Promise<ReadDirResult> = api.kernel
+    .getUserHomeDir()
+    .then((dir) => {
+      setWorkingDirectory(dir);
+      return dirPromise;
+    });
+</script>
+
+<Layout>
+  <CenterFormPanel title={`New ${starter} project`} backRoute="#/new-project">
+    <form
+      on:submit|preventDefault={async () => {
+        error = null;
+        let workspaceId;
+
+        if (starter !== 'empty') {
+          templateUrl = String(
+            (await api.kernel.describeTemplates()).find(
+              (x) => x.name === starter,
+            )?.url,
+          );
+        }
+
+        try {
+          workspaceId = await api.kernel.createProject(
+            `${workingDirectory}/${name}`,
+            templateUrl,
+          );
+          router.push(`/workspaces/${encodeURIComponent(workspaceId)}`);
+        } catch (ex) {
+          if (!isClientError(ex)) {
+            throw ex;
+          }
+          error = ex;
+        }
+      }}
+    >
+      <h1>New {starter} project</h1>
+
+      <label for="name">Name:</label>
+      <Textbox
+        bind:value={name}
+        name="name"
+        id="name"
+        placeholder={starter}
+        --input-width="100%"
+      />
+      <div style="height:32px" />
+      {#if workingDirectory}
+        <label for="root">Root:</label>
+        <h2><span>{workingDirectory}<span>/{name}</span></span></h2>
+        {#await dirPromise}
+          <Spinner />
+        {:then dir}
+          {#if dir.parent !== null}
```

Comment:
Minor, but I'd prefer disabling the button rather than removing it. Otherwise you have what happened to me where I was trying to get to the root directory by clicking the `..` button a bunch of times and I got to `/app` because when I got to the root directory the `..` button disappeared and `/app` appeared unexpectedly under my cursor.

Reply:
Done